### DISCUSSION TITLE
Going back from a favorite event to the schedule

### DISCUSSION THREAD

**Discussion by steelman** [EventFahrplan/EventFahrplan#380] [Show and tell]
# Environment
- App version: 1.49.1-FOSDEM-Edition (f-droid)
- Android version: LineageOS 17.1 (Android 10)
- Branch:
- Device: river

# How to reproduce
1. In the schedule screen choose favorites in the upper-right corner.
2. In the favorites list choose an event.
3. Touch (click) the arrow in the upper-left corner.

# Observed behavior
- The initial schedule screen appears

# Expected behavior
- The favorites list appears. This happens when `back` button on the bottom of the screen is used.

# Related
- #9

**Comment by johnjohndoe** [comment]
Thank you for submitting the issue. 

The behavior you observed is actually intended. The arrow in the upper-left (`home` button) always takes you to the main schedule screen. The system `back` button takes you back one step. The `home` button is just a shortcut which might be used from time to time. As the `back` button is way better to reach with the thumb it is the intended navigation button most of the time.

I hope my answer clarifies the case. Feel free to reply or close the issue.

**Comment by steelman** [comment]
If that is a feature, I am fine. However, on a slightly bigger screen when holding the device in one hand and operating with the other, after touching the bell icon on the toolbar it was easier (if only slightly) for me to reach for the `home` button. It frustrated me until by accident (as always) I discovered that the `back` button does what I need. So it's more about discoverability than the behavior itself.

_Maybe_ the first time users use `back` and `home` buttons in this scenario present them a tip-of-the-day dialogue telling about the other option. I supposed there could be more such tips around the app. Especially one like this which is used rather seldom.

**Comment by johnjohndoe** [comment]
I am totally able to relate to the observations you made. Thank you for writing it down here. Let me take your findings and put them into a temporal perspective.

At the time when the toolbar was implemented (~2011) it was common practice to put all the icons to the top and the rest behind the overflow menu. Later a side drawer menu was used in many apps. That was a time when smartphones were smaller and maybe less device sizes existed. 
Fast forward to today there is a variety of devices in all form factors. Also design guidelines have been [modernized](https://material.io/components/app-bars-bottom/android#bottom-app-bar). I believe it is about time to rethink the interaction options. I do not think a good design needs tool tips or tip-of-the-day dialogues - for me, it indicates that the user interface is not intuitive. When one revises the interaction options there is a good opportunity to consider how far users can reach with their thumb on different devices. Further, less frequently used actions probably do not need to be as reachable as the ones which are tapped all the time.

I have thought about various tasks in this area but I have not found time to work on it nor to write down a planning issue. I believe what it needs is a intense conversation up-front about different solutions, other parts of the app which are affected and visual mockups similar as they have been done by @hramberger in #232 (for another design topic). And once a good idea has been found somebody needs to implement it.
